src/datomish/sqlite.cljc

```diff
@@ -50,6 +50,15 @@
           [nil err]
           [@acc nil])))))
 
+(defn <all-rows
```

Comment:
I'm confused by your use of the name `pair-chan` there: `-each` calls its callback with just the row. The result of `reduce-rows` is a channel that results in a pair (I think?), but `pair-chan` in your example is a channel of rows.

Besides, `reduce-rows` uses an `atom`. Not only do we not need that — a channel is a queue; we don't need another mechanism for collection — but `swap!` can result in the same item being put into the channel multiple times. It's not safe to supply a side-effecting function as the argument to `swap!`.